kernel/packages/shared/apis/SceneStateStorageController/ISceneStateStorageController.ts

```diff
@@ -8,4 +8,5 @@ export interface ISceneStateStorageController {
   getProjectManifestByCoordinates(land: string): Promise<SerializedSceneState | undefined>
   createProjectWithCoords(coordinates: string): Promise<boolean>
   createProjectFromStateDefinition(): Promise<SerializedSceneState | undefined>
+  saveSceneInfo(sceneState: SerializedSceneState, sceneName: string, sceneDescription: string, sceneScreenshot: string): Promise<boolean>
```

Comment:
we already have the `saveSceneState` method. why not using that instead of creating a new one?

Reply:
`saveSceneState` and `saveSceneInfo` do different things and are called in different situations:
- `saveSceneState`: updates the entities states of the project manifest and is called each time a change is done in the BiW editor.
- `saveSceneInfo`: updates the basic info of the project manifest (name, description and screenshot) and is called only in 3 situation: when we create a new project, when we exit BiW editor and there have been changes or when we publish.

Reply:
as far as I can see. we will be trying to save the project twice to the builder-server in the case of publishing.
and we might find some unexpected results depending in which event comes first cause one will overwrite the other.
maybe we can send those payloads you need as optional and inject them to the project manifest if they exist instead of creating new events and methods? I don't know. What's Chamo's opinion?

Reply:
If you refer to 

`await this.builderApiManager.updateProjectManifest(builderManifest, this.getIdentity())`  

and 

`await this.builderApiManager.updateProjectThumbnail(builderManifest.project.id, thumbnailBlob, this.getIdentity())<!--EndFragment-->`

This should be done in the same call or at least use the same project and manifest since as stated by pbosio, it can cause some info lose in certain scenarios

Reply:
Personally I don't like the idea of reusing a same method with different info injected and, depending on if some of this info is null or not, follow a flow or other inside the method. For me each method should have only one responsibility. Are we having different methods using a common behaviour (in this case `updateProjectDetails()`? Yes, but I don't think it is a problem.

About what you say of the `publishSceneState`, it is not saving the project twice, it is saving the project in builder on the one hand and deploying it in server on the other. It is something that I already had implemented before this PR and was approved by Nico.

Anyway let ask @nchamo and see what he think. Thanks for the feedback Pato!

Reply:
@Suduck why it could cause info lose? We are doing the calls in this way:
```
await this.builderApiManager.updateProjectManifest(builderManifest, this.getIdentity())
await this.builderApiManager.updateProjectThumbnail(builderManifest.project.id, thumbnailBlob, this.getIdentity())
```

The second call is wating for the first one is finished.

Reply:
Yep, my bad I thought it was different

Reply:
I want to check if I understood correctly. 
Both `saveSceneState` and the new `saveProjectInfo` update the project on to Builder, right?

Reply:
I want to check if I understood correctly. 
Both `saveSceneState` and the new `saveProjectInfo` update the project on to Builder, right?

Reply:
I want to check if I understood correctly. 

Both `saveSceneState` and the new `saveProjectInfo` update the project on to Builder, right? And when they do it, they both send the scene's state to the Builder, right?

Now, my question is: is there an action (for example publishing) that would call both `saveSceneState` and the new `saveProjectInfo` together? Or are they always called at different times?

Reply:
@nchamo exactly, both `saveSceneState` and the new `saveProjectInfo` update the project on to Builder. Currently there not exist any action that is calling both functions together.

Reply:
Then I see no chance of a race condition that would cause problems. So it's a matter of how you think would be the best way to code it. I'll leave that for you to discuss and decide, since this whole code is something you've been working on for the last months 😊 

I'll take a look at the rest of the PR to check if I see something weird

Reply:
Ok, thanks Nico!